include/deal.II/non_matching/mapping_info.h

```diff
@@ -157,30 +326,544 @@ namespace NonMatching
+            const auto it_end =
+              unit_points.begin() + unit_points_index[current_face_index + 1];
+            return make_array_view(it_begin, it_end);
+          }
+        else
+          {
+            const unsigned int current_face_index =
+              cell_index_offset[cell_index] + face_number;
+            const auto it_begin =
+              unit_points.begin() + unit_points_index[current_face_index];
+            const auto it_end =
+              unit_points.begin() + unit_points_index[current_face_index + 1];
+            return make_array_view(it_begin, it_end);
+          }
+      }
+    else
+      AssertThrow(
+        false,
+        ExcMessage(
+          "cell_index has to be specified if face number is specified"));
+  }
+
+
+
+  template <int dim, int spacedim>
+  const typename MappingInfo<dim, spacedim>::MappingData &
+  MappingInfo<dim, spacedim>::get_mapping_data(
+    const unsigned int cell_index,
+    const unsigned int face_number) const
+  {
+    if (cell_index == numbers::invalid_unsigned_int &&
+        face_number == numbers::invalid_unsigned_int)
+      {
+        Assert(state == State::single_cell,
+               ExcMessage(
+                 "This mapping info is not reinitialized for a single cell"));
+        return mapping_data[0];
+      }
+    else if (face_number == numbers::invalid_unsigned_int)
+      {
+        Assert(state == State::cell_vector,
+               ExcMessage(
+                 "This mapping info is not reinitialized for a cell vector"));
+        if (do_cell_index_compression)
+          {
+            Assert(
+              cell_index_to_compressed_cell_index[cell_index] !=
+                numbers::invalid_unsigned_int,
+              ExcMessage(
+                "Mapping info object was not initialized for this active cell "
+                "index"));
+            return mapping_data
+              [cell_index_to_compressed_cell_index[cell_index]];
+          }
+        else
+          return mapping_data[cell_index];
+      }
+    else if (cell_index != numbers::invalid_unsigned_int)
+      {
+        Assert(state == State::faces_on_cells_in_vector,
+               ExcMessage("This mapping info is not reinitialized for faces"
+                          " on cells in a vector"));
+        if (do_cell_index_compression)
+          {
+            Assert(
+              cell_index_to_compressed_cell_index[cell_index] !=
+                numbers::invalid_unsigned_int,
+              ExcMessage(
+                "Mapping info object was not initialized for this active cell index"
+                " and corresponding face numbers"));
+            return mapping_data
+              [cell_index_offset
+                 [cell_index_to_compressed_cell_index[cell_index]] +
+               face_number];
+          }
+        else
+          return mapping_data[cell_index_offset[cell_index] + face_number];
+      }
+    else
+      AssertThrow(
```

Comment:
I think we can convert most `AssertThrow` instances to `Assert.`

Reply:
If I change this I have to return something for this invalid case.